plugin/pkg/admission/podnodeselector/admission.go

```diff
@@ -114,20 +115,23 @@ func (p *podNodeSelector) Admit(a admission.Attributes) error {
 
 	resource := a.GetResource().GroupResource()
 	pod := a.GetObject().(*api.Pod)
-	namespaceNodeSelector, err := p.getNamespaceNodeSelectorMap(a.GetNamespace())
+
+	var namespaceNodeSelector labels.Set
+	namespaceNodeSelector, err = p.verifyConflictWithNamespaceNodeSelectors(resource, pod.Name, pod.Namespace, pod.Spec.NodeSelector)
 	if err != nil {
 		return err
 	}
 
-	if labels.Conflicts(namespaceNodeSelector, labels.Set(pod.Spec.NodeSelector)) {
```

Comment:
Why remove this? It's better to reject early.

Reply:
Actually in the code flow with verifyConflictWithNamespaceNodeSelectors, labels.Conflicts is exactly the same place. isn't it?